[build] - Update build scripts to support llvm install into lib/llvm

estewart08 opened the pull request:
Introduces LLVM_INSTALL_LOC to avoid code duplication in various scripts.

estewart08 committed:
[build] - Update build scripts to support llvm install into lib/llvm

Introduces LLVM_INSTALL_LOC to avoid code duplication
in various scripts.

estewart08 commented:
This is reliant on a llvm patch, they will both have to be merged at the same time to prevent breakage. There is still a runtime issue with using lib-debug libraries. Not ready to merge yet.

ronlieb reviewed (commented):
i assume you used patch to build and it works?

should wait for Greg to review.

when do we want to roll this out ?

estewart08 committed:
[build] - Updates for amdsmi, rocm_smi_lib and rocr

More prep for llvm install location lib/llvm.

estewart08 committed:
[build][smoke] - Additional changes for moving llvm install

estewart08 committed:
[build] - Ensure gpurun installs into llvm/bin

estewart08 committed:
Additional updates for lib/llvm support.

Adjusted a few smoke-dev/smoke-fails/smoke Makefiles.
Added hipamd patch to override RPATH.
Add config file creation in build_project.sh which aids
hippc compilation.
